File: app/src/test/java/vocabletrainer/heinecke/aron/vocabletrainer/FormatterTest.java
package vocabletrainer.heinecke.aron.vocabletrainer;

import org.junit.Test;

import vocabletrainer.heinecke.aron.vocabletrainer.lib.Formatter;

import static org.junit.Assert.assertEquals;

public class FormatterTest {
    @Test
    public void format_isCorrect() {
        Formatter fmt = new Formatter();
        fmt.changeSI(true);
        long bytes_1024 = 1024;
        long bytes_mib = 1048576;
        long bytes_1000 = 1000;
        long bytes_mb = 1000000;
        assertEquals("1 KiB", fmt.formatBytes(bytes_1024));
        assertEquals("1 MiB", fmt.formatBytes(bytes_mib));
        assertEquals("1,000 B", fmt.formatBytes(bytes_1000));
        assertEquals("976.6 KiB", fmt.formatBytes(bytes_mb));
    }
}

File: app/src/test/java/vocabletrainer/heinecke/aron/vocabletrainer/FormatterTest.kt
package vocabletrainer.heinecke.aron.vocabletrainer

import org.junit.Assert
import org.junit.Test
import vocabletrainer.heinecke.aron.vocabletrainer.lib.Formatter
import java.text.DecimalFormatSymbols

class FormatterTest {
    @Test
    fun format_isCorrect() {
        val fmt = Formatter()
        fmt.changeSI(true)
        val bytes_1024: Long = 1024
        val bytes_mib: Long = 1048576
        val bytes_1000: Long = 1000
        val bytes_mb: Long = 1000000
        Assert.assertEquals("1 KiB", fmt.formatBytes(bytes_1024))
        Assert.assertEquals("1 MiB", fmt.formatBytes(bytes_mib))
        val dfs = DecimalFormatSymbols.getInstance()
        Assert.assertEquals("1${dfs.groupingSeparator}000 B", fmt.formatBytes(bytes_1000))
        Assert.assertEquals("976${dfs.decimalSeparator}6 KiB", fmt.formatBytes(bytes_mb))
    }
}